src/Services/PopoverManager.vala

```diff
@@ -102,6 +102,27 @@ public class Wingpanel.Services.PopoverManager : Object {
             Gtk.Allocation allocation;
             popover.get_allocation (out allocation);
 
+            Gtk.Allocation indicator_allocation;
+            current_indicator.get_allocation (out indicator_allocation);
+
+            Gtk.Allocation container_allocation;
+            current_indicator.get_parent ().get_allocation (out container_allocation);
+
+            int wingpanel_width;
+            current_indicator.get_root_window ().get_geometry (null, null, out wingpanel_width, null);
```

Comment:
Can't we just use the already available `owner` field here?

Reply:
Thanks for the review, @donadigo 

Since `get_root_window` returns a `Gdk.Window`, not a `Gtk.Window`. We still need the call to `get_root_window`. However, we still get the same resulting window if we get it through the `owner` and it may be slightly more efficient, so I've push a commit that switches to that.

Thanks.